spec/std/dir_spec.cr

```diff
@@ -179,6 +179,18 @@ describe "Dir" do
       ].sort
     end
 
+    it "tests double recursive matcher" do
+      with_tempfile "glob-double-recurse" do |path|
+        sub_4 = Path[path, "sub-1", "sub-2", "sub-3", "sub-4"]
+        Dir.mkdir_p sub_4
+        File.touch sub_4.join("file")
+        Dir["#{path}/sub-1/**/sub-3/**/*"].sort.should eq [
+          sub_4.to_s,
+          sub_4.join("file").to_s
+        ].sort
+      end
+    end
+
```

Comment:
I'd add at least a test that it *doesn't* match something that *doesn't* contain `/sub-3/`

Reply:
Precision isn't really an issue tho. The way the algorithm works, it can only yield existing paths as results.
The problems are missing results (which this patch should correct) and duplicate results (which this patch introduces).